src/main/java/com/o19s/es/ltr/feature/store/PrecompiledExpressionFeature.java

```diff
@@ -86,8 +125,8 @@ public int hashCode() {
 
     @Override
     public void validate(FeatureSet set) {
-        for(String var : expression.variables) {
-            if(!set.hasFeature(var)) {
+        for (String var : expression.variables) {
+            if (!set.hasFeature(var) && !queryParams.contains(var)) {
                 throw new IllegalArgumentException("Derived feature [" + this.name + "] refers " +
                         "to unknown feature: [" + var + "]");
```

Comment:
nitpick: `refers to unknown feature or parameter`

Reply:
fixed